### PR TITLE
move some logic out of success/failure condition

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Beginning of the fix for https://www.pivotaltracker.com/story/show/91023678 which also requires LB changes.

Currently the code on LB is:
Success

```
function () {
  if (!Studio.__hacked) {
    Studio.__hacked = true;
  }
  var originalOnTick = Studio.customLogic.constructor.prototype.onTick.bind(Studio.customLogic);
  Studio.customLogic.constructor.prototype.onTick = function() {
    if (this.studio_.tickCount === 1) {
      this.seconds = 0;
      this.height = 0;
    }
    originalOnTick();
    Studio.Globals.time = this.seconds;
    Studio.Globals.succeeded = ((this.seconds == 21 && this.height == 315) && (this.rocket_height(10) == 150));
  }
  return Studio.Globals.succeeded;
}
```

Failure

```
function () {
  return (Studio.Globals.time > 21 && !Studio.Globals.succeeded)
}
```

I will be updating it to:
Success:

```
function () {
  var seconds = Studio.customLogic.seconds;
  var height = Studio.customLogic.height;
  var SECONDS_TO_RUN = Studio.customLogic.SECONDS_TO_RUN;
  var targetHeight = SECONDS_TO_RUN * 15;

  // If at target time, make sure we have the right height for this time and
  // the next time.
  return seconds === SECONDS_TO_RUN && height === targetHeight &&
    Studio.customLogic.rocket_height(SECONDS_TO_RUN + 1) === targetHeight + 15;
}
```

Failure:

```
function () {
  return Studio.customLogic.seconds > Studio.customLogic.SECONDS_TO_RUN;
}
```

### CODE REVIEW

File: apps/src/studio/rocketHeightLogic.js [hunk 1]
```diff
@@ -17,10 +17,21 @@ var RocketHeightLogic = function (studio) {
...
   this.rocket = null;
   this.height = 0;
+
+  // Use by successCondition/failureCondition
+  this.SECONDS_TO_RUN = 8;
 };
 RocketHeightLogic.inherits(CustomGameLogic);
 
 RocketHeightLogic.prototype.onTick = function () {
+  if (this.studio_.tickCount === 1) {
```
Comment (bcjordan) [inline apps/src/studio/rocketHeightLogic.js]: Is this field 1-indexed? Or this doesn't get called first tick?

Comment (bcjordan) [inline apps/src/studio/rocketHeightLogic.js]: Is this field 1-indexed? Or this doesn't get called first tick?

Comment (bcjordan) [inline apps/src/studio/rocketHeightLogic.js]: Is this field 1-indexed? Or this doesn't get called first tick?

Comment (Bjvanminnen) [inline apps/src/studio/rocketHeightLogic.js]: It gets called after the tick is incremented (so the first tick increments 0 to 1, then calls this).

Comment (bcjordan) [inline apps/src/studio/rocketHeightLogic.js]: Sorry for multiple comments, connection struggling :)

**Comment by bcjordan** [comment]
LGTM

**Pull request merged**